ui/components/schema/SchemaContainer.js

```diff
@@ -19,16 +19,22 @@ class SchemaContainer extends Component {
     constructor(props) {
         super(props);
         this.state = INITIAL_STATE;
+        this.flexWrapper = React.createRef();
     }
 
     updateDimensions() {
         this.setState({
-            height: window.innerHeight - this.calculateHeaderHeight()
+            height: window.innerHeight - this.flexWrapper.current.getBoundingClientRect().top
         });
     }
 
-    componentWillMount() {
-        this.updateDimensions();
+    componentDidUpdate() {
+        const flexWrapperElement = this.flexWrapper.current;
+        const currentHeight = flexWrapperElement.style.height;
+        const distanceFromTop = flexWrapperElement.getBoundingClientRect().top;
+        if (currentHeight === "0px" && distanceFromTop !== 0) {
```

Comment:
Without checking the initial state (`currentHeight === "0px"`) it would end up calling `this.updateDimensions()` indefinitely (because  each update would trigger `componentDidUpdate()` method and the condition `distanceFromTop !== 0` would be always true)